Detectors/TOF/base/include/TOFBase/CalibTOFapi.h

```diff
@@ -107,17 +117,17 @@ class CalibTOFapi
   Diagnostic* mDiaFreq = nullptr;  ///< object for Diagnostic Frequency
 
   // info from diagnostic
-  int mNoisyThreshold = 1;                          ///< threshold to be noisy
-  float mEmptyTOF = 0;                              ///< probability to have TOF fully empty
-  float mEmptyCrateProb[Geo::kNCrate];              ///< probability to have an empty crate in the current readout window
-  std::vector<std::pair<int, float>> mNoisy;        ///< probTRMerror
-  std::vector<std::pair<int, float>> mTRMerrorProb; ///< probTRMerror
-  std::vector<int> mTRMmask;                        ///< mask error for TRM
-
-  bool mIsErrorCh[Geo::NCHANNELS];  ///< channels in error (TRM)
-  std::vector<int> mFillErrChannel; ///< last error channels filled
-  bool mIsOffCh[Geo::NCHANNELS];    ///< channels in error (TRM)
-  bool mIsNoisy[Geo::NCHANNELS];    ///< noisy channels
+  int mNoisyThreshold = 1;                           ///< threshold to be noisy
+  float mEmptyTOF = 0;                               ///< probability to have TOF fully empty
+  float mEmptyCrateProb[Geo::kNCrate] = {INITF_ALL}; ///< probability to have an empty crate in the current readout window
+  std::vector<std::pair<int, float>> mNoisy;         ///< probTRMerror
+  std::vector<std::pair<int, float>> mTRMerrorProb;  ///< probTRMerror
+  std::vector<int> mTRMmask;                         ///< mask error for TRM
+
+  bool mIsErrorCh[Geo::NCHANNELS] = {INITB_ALL}; ///< channels in error (TRM)
+  std::vector<int> mFillErrChannel;              ///< last error channels filled
+  bool mIsOffCh[Geo::NCHANNELS] = {INITB_ALL};   ///< channels in error (TRM)
+  bool mIsNoisy[Geo::NCHANNELS] = {INITB_ALL};   ///< noisy channels
```

Comment:
@noferini You did not have array initialization at all, otherwise it would work :)
The code like 
```
float mEmptyCrateProb[Geo::kNCrate];
```
leaves array non-initilized. To set all elements to 0 it is enough to do
```
float mEmptyCrateProb[Geo::kNCrate] = {};
```
The same for all arrays you have (also the std::arrays)

Reply:
I did a memset in the constructor

Reply:
and it worked

Reply:
and it still works outside the constructor

Reply:
After email discussion the bug was understood (it was a my fault in previous implementation)
Thanks!